src/Bech32Address.cpp

```diff
@@ -45,7 +45,13 @@ bool Bech32Address::decode(const std::string& addr, Bech32Address& obj_out, cons
     }
 
     Data conv;
-    auto success = Bech32::convertBits<5, 8, false>(conv, std::get<1>(dec));
+    bool success;
+    if (pad) {
+        // cannot supply function arguments to template parameters, will cause compiler error
+        success = Bech32::convertBits<5, 8, true>(conv, std::get<1>(dec));
```

Comment:
I've tried taking out this if, and all tests passed.
Please provide a test case where padding makes a difference, or remove the padding option.